tmt/utils/git.py

```diff
@@ -0,0 +1,605 @@
+    re_source = re.compile(r"^(\w+) \(([^)]+)\) = ([0-9a-fA-F]+)$")
+    lookaside_server = "https://src.fedoraproject.org/repo/pkgs"
+    remote_substring = re.compile(r'fedoraproject\.org')
+
+
+class CentOSDistGit(DistGitHandler):
+    """ CentOS Handler """
+
+    usage_name = "centos"
+    re_source = re.compile(r"^(\w+) \(([^)]+)\) = ([0-9a-fA-F]+)$")
+    lookaside_server = "https://sources.stream.centos.org/sources"
+    remote_substring = re.compile(r'redhat/centos')
+
+
+class RedHatGitlab(DistGitHandler):
+    """ Red Hat on Gitlab """
+
+    usage_name = "redhatgitlab"
+    re_source = re.compile(r"^(\w+) \(([^)]+)\) = ([0-9a-fA-F]+)$")
+    # Location already public (standard-test-roles)
+    lookaside_server = "http://pkgs.devel.redhat.com/repo"
+    remote_substring = re.compile(r'redhat/rhel/')
+
+
+def get_distgit_handler(
+        remotes: Optional[list[str]] = None,
+        usage_name: Optional[str] = None) -> DistGitHandler:
+    """
+    Return the right DistGitHandler
+
+    Pick the DistGitHandler class which understands specified
+    remotes or by usage_name.
+    """
+    for candidate_class in DistGitHandler.__subclasses__():
+        if usage_name is not None and usage_name == candidate_class.usage_name:
+            return candidate_class()
+        if remotes is not None:
+            ret_val = candidate_class()
+            if ret_val.its_me(remotes):
+                return ret_val
+    raise GeneralError(f"No known remote in '{remotes}'.")
+
+
+def get_distgit_handler_names() -> list[str]:
+    """ All known distgit handlers """
+    return [i.usage_name for i in DistGitHandler.__subclasses__()]
+
+
+def distgit_download(
+        *,
+        distgit_dir: Path,
+        target_dir: Path,
+        handler_name: Optional[str] = None,
+        caller: Optional['Common'] = None,
+        logger: tmt.log.Logger
+        ) -> None:
+    """
+    Download sources to the target_dir
+
+    distgit_dir is path to the DistGit repository
+    """
+    # Get the handler unless specified
+    if handler_name is None:
+        cmd = Command("git", "config", "--get-regexp", '^remote\\..*.url')
+        output = cmd.run(cwd=distgit_dir,
+                         caller=caller,
+                         logger=logger)
+        if output.stdout is None:
+            raise tmt.utils.GeneralError("Missing remote origin url.")
+        remotes = output.stdout.split('\n')
+        handler = tmt.utils.get_distgit_handler(remotes=remotes)
+    else:
+        handler = tmt.utils.get_distgit_handler(usage_name=handler_name)
+
+    for url, source_name in handler.url_and_name(distgit_dir):
+        logger.debug(f"Download sources from '{url}'.")
+        with tmt.utils.retry_session() as session:
+            response = session.get(url)
+        response.raise_for_status()
+        target_dir.mkdir(exist_ok=True, parents=True)
+        with open(target_dir / source_name, 'wb') as tarball:
+            tarball.write(response.content)
```

Comment:
This is not a suggestion, as this is not a new code, only moved from the other file. (and wouldn't work as is) 
I'd just like to ask for guidelines for situations like this, where I notice something unimportant, trivial, not relevant to the PR.     
@happz What would you want me to do, open an issue with something like "Use Pathlib when appropriate"? I assume I should not mention these in the reviews to keep things on point.. 
```suggestion
        Path(target_dir / source_name).write_bytes(response.content)`
```

Reply:
The best way would be, of course, submitting a PR fixing said minor issue :) An issue seems like an overkill to me, unless you're not sure how to fix it or there are some side-effects to discuss deeper.